elasticapm/handlers/logging.py

```diff
@@ -155,3 +156,51 @@ def _emit(self, record, **kwargs):
             logger_name=record.name,
             **kwargs
         )
+
+
+class LoggingFilter(logging.Filter):
+    """
+    This filter doesn't actually do any "filtering" -- rather, it just adds
+    two new attributes to any "filtered" LogRecord objects:
+
+    * elasticapm_transaction_id
+    * elasticapm_transaction_trace_parent_id
+
+    These attributes can then be incorporated into your handlers and formatters,
+    so that you can tie log messages to transactions in elasticsearch.
+    """
+
+    def filter(self, record):
+        """
+        Add elasticapm attributes to `record`.
+        """
+        transaction = execution_context.get_transaction()
+        record.elasticapm_transaction_id = transaction.id
+        record.elasticapm_transaction_trace_parent_id = (
+            transaction.trace_parent.trace_id if transaction.trace_parent else None
+        )
+        return True
+
+
+def structlog_processor(_, __, event_dict):
+    """
+    Add two new entries to the event_dict for any processed events:
+
+    * elasticapm_transaction_id
+    * elasticapm_transaction_trace_parent_id
+
+    :param _:
```

Comment:
I'd still use the argument names as they seem to be convention in [structlog](http://www.structlog.org/en/0.4.0/processors.html), even if they are unused.

Reply:
I skipped them only because in `structlog`'s examples, they use these same names when the underlying processor doesn't care about them. I can name them as what they are, I'm not picky here.

Reply:
Sorry, should have left a link: http://www.structlog.org/en/stable/examples.html#processors

Note that their sample processors use the same convention.

Reply:
Oh, didn't see that, interesting!

I've just never seen this convention before, using underscores as argument names (I've seen it for variable names in multiple assignments, something like, `val, _ = some_tuple`), and it's something I'd do a double take every time I happen to read it. You alleviate that somewhat by documenting what the arguments are in the docstring, but still.

Reply:
I just named them normally. I wasn't invested in either outcome. :)